network/lwip2transport/device.go

```diff
@@ -0,0 +1,191 @@
+// this device.
+func (d *lwIPDevice) MTU() int {
+	return packetMTU
+}
+
+// writeResponse writes an incoming IP packet response `b` to this device. The packet can be read by calling the [Read]
+// function, or it can be redirected to an [io.Writer] if the [WriteTo] function has been called. writeResponse blocks
+// until the packet is successfully consumed by a [Read] or [WriteTo].
+//
+// writeResponse can be used as an output function for lwIP.
+//
+// writeResponse might be called by multiple goroutines (for example, when multiple UDP packets arrive at the same
+// time). We sequentialize the calls by using channels, if performance issues arise in the future, we can use other
+// more performant but more error-prone methods (e.g. the [sync] package) to resolve them.
+func (d *lwIPDevice) writeResponse(b []byte) (int, error) {
+	select {
+	case d.rdBuf <- b:
+		select {
+		case n := <-d.rdN:
+			return n, nil
+		case <-d.done:
+			return 0, network.ErrClosed
+		}
+	case <-d.done:
+		return 0, network.ErrClosed
+	}
+}
+
+// Read implements [io.Reader] and [network.IPDevice]. It reads one IP packet from the TCP/UDP response, blocking until
+// a packet arrives or this device is closed. If a packet is too long to fit in the supplied buffer `p`, the excess
+// bytes are discarded.
+//
+// Read returns [io.EOF] error if this device is closed or no more data is available.
+func (d *lwIPDevice) Read(p []byte) (int, error) {
+	select {
+	case s := <-d.rdBuf:
+		n := copy(p, s)
+		d.rdN <- n
+		return n, nil
+	case <-d.done:
+		return 0, io.EOF
+	}
+}
+
+// WriteTo implements [io.WriterTo]. It writes all IP packets from TCP/UDP responses to `w` until all data is written
+// or an error occurs. This function will not allocate any intermediate buffers.
+//
+// WriteTo returns the total number of bytes written and any error encountered during the write. If there are no more
+// data available, WriteTo returns nil error instead of [io.EOF].
+func (d *lwIPDevice) WriteTo(w io.Writer) (int64, error) {
+	nw := int64(0)
+	for {
+		select {
+		case s := <-d.rdBuf:
+			n, err := w.Write(s)
+			nw += int64(n)
+			select {
+			case d.rdN <- n:
+				if err != nil {
+					return nw, err
+				}
+			case <-d.done:
+				return nw, nil
+			}
+		case <-d.done:
+			return nw, nil
+		}
+	}
+}
+
+// Write implements [io.Writer] and [network.IPDevice]. It writes a single IP packet to this device. The device will
+// then translate the IP packet into a TCP or UDP traffic.
+//
+// Write returns [network.ErrClosed] if this device is already closed.
+// Write returns [network.ErrMsgSize] if len(b) > [MTU].
+func (d *lwIPDevice) Write(b []byte) (int, error) {
+	select {
+	case <-d.done:
+		return 0, network.ErrClosed
+	default:
+	}
+	if len(b) > d.MTU() {
+		return 0, network.ErrMsgSize
+	}
+	n, err := d.stack.Write(b)
+	// workaround: lwip netstack did not use a standard error code
```

Comment:
Start with capital letter. And perhaps "typed error" is more accurate than standard error.
```suggestion
	// Workaround: lwip netstack did not use a typed error.
```